apps/system/js/notifications.js

```diff
@@ -122,6 +126,86 @@ var NotificationScreen = {
     }, function(err) {
       console.error('VersionHelper failed to lookup version settings.');
     });
+
+
+  },
+
+  getStore: function ns_getStore() {
+    return new Promise(resolve => {
+      if (this._storeRef) {
+        return resolve(this._storeRef);
+      }
+      navigator.getDataStores(this.STORE_NAME).then(stores => {
+        this._storeRef = stores[0];
+        return resolve(this._storeRef);
+      });
+    });
+  },
+
+  increaseAppCounter: function ns_udpateCounter(detail) {
+    this.debug('increaseAppCounter('+JSON.stringify(detail)+')');
+    this.debug('has url = ' + this._countersPerApp.hasOwnProperty(detail.manifestURL));
+    this._countersPerApp.hasOwnProperty(detail.manifestURL) ?
+      this._countersPerApp[detail.manifestURL] += 1 :
+      this._countersPerApp[detail.manifestURL] = 1;
+
+    function addCounter(store, name, count) {
+      store.add(count, name).then( id => {
+        this.debug('stored ' + id);
+      });
+    };
+
+    function updCounter(store, name, count) {
+      store.put(count, name).then( id => {
+        this.debug('updated ' + id);
+      });
+    };
+
+    this.getStore().then( store => {
+      store.getLength().then( storeLength => {
+        if(storeLength == 0) {
+          this.debug('empty store');
+          Object.getOwnPropertyNames(this._countersPerApp).forEach( item => {
+            addCounter.call(this, store, item, this._countersPerApp[item]);
+          });
+        } else {
+          this.debug('non empty store');
+          updCounter.call(this, store, detail.manifestURL, this._countersPerApp[detail.manifestURL]);
+        }
+      });
+    });
+  },
+
+  decreaseAppCounter: function ns_decAppCounter(appURL) {
+    // very rough implementation - easy to get negative count.
+    if (this._countersPerApp.hasOwnProperty(appURL)) {
+      this._countersPerApp[appURL] -= 1;
+    } else {
+      this.debug('WARNING trying to decrease counter for uncnown URL');
+      return;
+    }
+
+    this.getStore().then(store => {
+      if (this._countersPerApp[appURL] > 0) {
+        this.debug('non empty store');
+        store.put(this._countersPerApp[appURL], appURL).then( id => {
+          this.debug('decremented ' + id);
+        });
+      } else {
+        //do we need to delete URLs for which no notifications left?
+        store.remove(appURL).then( success => {
+          if (success) {
+            this.debug('deleted ' + appURL);
+          } else {
+            this.debug('Something went wrong with deletion of ' + appURL);
+          }
+        });
+      }
+    });
+  },
+
+  debug: function ns_debug(msg) {
+    console.log('[el] ' + msg);
```

Comment:
definitely, that's why in 'list of features/changes' in this PR description I put line number 7 - 'remove extra debugging information (mostly could be found by searching [el] pattern)' :)